src/mem/mediumslab.h

```diff
@@ -22,7 +22,7 @@ namespace snmalloc
     uint16_t free;
     uint8_t head;
     uint8_t sizeclass;
-    uint16_t stack[SLAB_COUNT - 1];
+    FreePtr<void> stack[SLAB_COUNT - 1];
```

Comment:
This looks as if it's a big increase in the metadata size for a medium slab (especially on CHERI, where each entry goes from 2 to 16 bytes).  Is it necessary?

Reply:
Good catch, this would push the meta-data size over a page.  I would be careful that there aren't corner cases that are bad here.  The increase is probably not going to affect much in terms of performance or memory usage.

Reply:
Ah, good catch re: crossing a page boundary.  I had converted these to `FreePtr<void>`s so that we didn't have to `ptrauth_bound` on the allocation path and so that, when the time comes, the version tags would be available implicitly... but
* the `CSetBounds` is probably not very expensive and
* it's might be worth instead having `uint16_t stack[SLAB_COUNT-1]` and `uint8_t stack_memver[SLAB_COUNT-1]`; this will necessitate further AAL methods for memory versioning, but that's probably fine.

If we do decide that `CSetBounds`+`CSetVersion` on the alloc path and `CGetVersion` on free is collectively too pricey, I wonder if it might be worth having a pair of SLAB_COUNT/2-sized stacks, one of which lives on a free page.  That'd keep everything small enough and shouldn't add much code complexity.  But that's a discussion for another PR, really.